### PR TITLE
enable_redis - Added redis cache hostname, port and pass

### PR DESCRIPTION
**Pull request by Thomas-Geraghty**
### Change description ###

- Added redis host, port and pass
- Added redis healtcheck

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/modules/properties-volume/index.ts [hunk 1]
```diff
@@ -12,6 +12,10 @@ export class PropertiesVolume {
       this.setSecret('secrets.idam-idam.AppInsightsInstrumentationKey', 'appInsights.instrumentationKey');
       this.setSecret('secrets.idam-idam.launchdarkly-sdk-key', 'launchdarkly.sdkKey');
       this.setSecret('secrets.idam-idam.idam-user-dashboard-client-secret', 'services.idam.clientSecret');
+      this.setSecret('secrets.idam-idam.redis-hostname', 'session.redis.host');
+      this.setSecret('secrets.idam-idam.redis-port', 'session.redis.port');
+      this.setSecret('secrets.idam-idam.redis-key', 'session.redis.key');
+      this.setSecret('secrets.idam-idam.redis-key', 'session.secret');
```
Comment (KianKwa) [inline src/main/modules/properties-volume/index.ts:18]: You are adding the redis-key as secret here....Is that intentional to use the key as secret?

File: src/main/modules/session/index.ts [hunk 2]
```diff
@@ -27,13 +27,14 @@ export class SessionStorage {
 
   private getStore(app: Application): session.Store {
     const redisHost: string = config.get('session.redis.host');
+    const redisPort: number = config.get('session.redis.port');
     const redisPass: string = config.get('session.redis.key');
 
-    if (redisHost && redisPass) {
+    if (redisHost && redisPort && redisPass) {
```
Comment (KianKwa) [inline src/main/modules/session/index.ts:33]: Why do we need to check the redisPort in the condition? Further down you are doing a check fir null too "redisPort ?? 6380"

File: src/main/modules/properties-volume/index.ts (hunk 1 above)
Comment (Thomas-Geraghty) [reply src/main/modules/properties-volume/index.ts:18]: Yep, seems common place (at least in HMCTS) to use it as the session secret

File: config/custom-environment-variables.yaml [hunk 3]
```diff
@@ -14,9 +14,9 @@ services:
     scope: 'openid profile roles manage-user search-user'
     responseType: 'code'
 session:
-  secret: 'local'
```
Comment (KianKwa) [inline config/custom-environment-variables.yaml:17]: THis config setting is removed here but think it is still used in the 'property-volume' index.ts and 'session' index.js

File: config/custom-environment-variables.yaml [hunk 4]
```diff
@@ -14,9 +14,9 @@ services:
...
   redis:
     host: REDIS_HOST
+    port: REDIS_PORT
```
Comment (KianKwa) [inline config/custom-environment-variables.yaml:19]: Does this new config need to be in default.yaml too?

File: src/main/modules/session/index.ts (hunk 2 above)
Comment (Thomas-Geraghty) [reply src/main/modules/session/index.ts]: Fixed in https://github.com/hmcts/idam-user-dashboard/pull/108/commits/5885dfa7362f02f8982715bacc8cbe1a9b942454

File: config/custom-environment-variables.yaml (hunk 4 above)
Comment (Thomas-Geraghty) [reply config/custom-environment-variables.yaml:19]: Fixed in https://github.com/hmcts/idam-user-dashboard/pull/108/commits/64b356222517afd091e37ed3d3a332a88fc7869d

**Review by KianKwa — APPROVED** [review]

**Pull request merged**